src/cargo/core/features.rs

```diff
@@ -40,9 +40,41 @@
 //! we'll be sure to update this documentation!
 
 use std::env;
+use std::fmt;
+use std::str::FromStr;
 
 use util::errors::CargoResult;
 
+/// The epoch of the compiler (RFC 2052)
+///
+/// Pronounced e-poach
+#[derive(Clone, Copy, Debug, Hash, PartialOrd, Ord, Eq, PartialEq)]
+pub enum Epoch {
+    /// The 2015 epoch
+    Epoch2015 = 2015,
```

Comment:
Cute, but isn't the only effect of this to make this enum a u16?

Reply:
It's more there for forcing the PartialOrd to be the right order. But it's not necessary for Cargo; I just copied this definition from the Rust PR (where comparing epochs is more important)